_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_decoder.cpp

```diff
@@ -486,7 +495,7 @@ namespace UMC_MPEG2_DECODER
             auto umcRes = StartFrame(slice);
             if (umcRes != UMC::UMC_OK || !m_currFrame)
```

Comment:
@dmitryermilov ,
If _StartFrame_ may return an error only when we have no free frame (and this means _m_currFrame_ is null) why do we need to check both - error result and _m_currFrame_? I'd rather make _StartFrame_ to return _MPEG2DecoderFrame_ pointer and use it as `m_currFrame = StartFrame(slice)`.

Reply:
@dmitry-gurulev , thanks ! I'll address this in another PR